lib/services/message-service.ts

```diff
@@ -0,0 +1,198 @@
+import * as Promise from 'bluebird';
+import * as bodyParser from 'body-parser';
+import * as express from 'express';
+import { Worker } from '../framework/worker';
+import { WorkerClient } from '../framework/worker-client';
+import {
+    Logger,
+    LogLevel,
+} from '../utils/logger';
+import {
+    MessageEmitResponse,
+    MessageEvent,
+    MessageWorkerEvent,
+    ReceiptContext,
+} from '../utils/message-types';
+import {
+    ServiceEmitContext,
+    ServiceEmitRequest,
+    ServiceEmitter,
+    ServiceListener,
+    ServiceRegistration,
+} from './service-types';
+
+/**
+ * Abstract class to define a common set of utilities and standards for all messenger classes
+ */
+export abstract class MessageService
+extends WorkerClient<string|null>
+implements ServiceListener, ServiceEmitter {
+    protected static logger = new Logger();
+    private static _app: express.Express;
+    private listening: boolean = false;
+    private _eventListeners: { [event: string]: ServiceRegistration[] } = {};
+
+    /**
+     * All messenger classes share a single express instance
+     */
+    protected static get app(): express.Express {
+        if(!MessageService._app) {
+            MessageService._app = express();
+            MessageService._app.use(bodyParser.json());
+            if(process.env.PORT) {
+                MessageService._app.listen(process.env.PORT);
```

Comment:
Should at least be `MESSAGE_SERVICE_PORT`. I'm really not convinced this should be a single static instance, in case, for example, bots specifically want to use different ports for different services (for reasons such as load balancing, firewalling, etc). I'd far prefer to see a per-service instance which allow passing a port.